gh-125331: Allow the parser to activate future imports on the fly (#125482)
python/cpython: Misc/NEWS.d/next/Core_and_Builtins/2024-10-29-23-30-35.gh-issue-125331.quKQ7V.rst

Using ``from __future__ import barry_as_FLUFL`` works now when using

> multiple statements on the same line. Additionally the effect is respected
> even if the import is used even if the flags are not passed to

Reviewer: ```suggestion
multiple statements on the same line. Additionally, the effect is respected
even if the import is used but flags are not passed to
```

> :func:`compile`. Patch by Pablo Galindo

Reviewer 2: ```suggestion
:func:`compile`. Patch by Pablo Galindo.
```

> Using ``from __future__ import barry_as_FLUFL`` works now when using
> multiple statements on the same line. Additionally, the effect is respected
> even if the import is used but flags are not passed to
> :func:`compile`. Patch by Pablo Galindo.

Reviewer 3: This actually undersells the things this fixes:
```suggestion
``from __future__ import barry_as_FLUFL`` now works in more contexts,
including when it is used in files, with the ``-c`` flag, and in the REPL
when there are multiple statements on the same line. Previously, it worked
only on subsequent lines in the REPL, and when the appropriate flags were
passed directly to :func:`compile`. Patch by Pablo Galindo.
```
